src/MainWindow.vala

```diff
@@ -32,9 +33,14 @@ public class Dock.MainWindow : Gtk.ApplicationWindow {
         child = box;
         overflow = Gtk.Overflow.VISIBLE;
         resizable = false;
+        decorated = false;
```

Comment:
Unfortunately it didn't fix the glitches :(
Remove it please. New solution will have to be found later.